### PR TITLE
Assert machine_type is set to q35

### LINKED ISSUE #732: add test coverage for q35 machine type by defualt

### PR DESCRIPTION
**Pull request by auniyal61**
Closed #732

### CODE REVIEW

File: test/functional/nova_reconfiguration_test.go [hunk 1]
```diff
@@ -790,4 +790,23 @@ var _ = Describe("Nova reconfiguration", func() {
 			g.Expect(diff).To(BeEmpty())
 		}, timeout, interval).Should(Succeed())
 	})
+
+	When("nova compute is recognized", func() {
+		It("machine_type is set properly", func() {
+			// this test read a nova.conf for nova compute and verify machine_type is set properly
```
Comment (mrkisaolamb) [inline test/functional/nova_reconfiguration_test.go:796]: I'm not sure if we need these comments. I would suggest either changing them to describe the reasoning behind the hw_machine_type setting or simply deleting them.

File: test/functional/nova_reconfiguration_test.go [hunk 2]
```diff
@@ -790,4 +790,23 @@ var _ = Describe("Nova reconfiguration", func() {
 			g.Expect(diff).To(BeEmpty())
 		}, timeout, interval).Should(Succeed())
 	})
+
+	When("nova compute is recognized", func() {
```
Comment (mrkisaolamb) [inline test/functional/nova_reconfiguration_test.go:794]: this test looks more like part of generall nova compute testing: https://github.com/openstack-k8s-operators/nova-operator/blob/main/test/functional/nova_compute_ironic_controller_test.go#L168

**Review by mrkisaolamb — CHANGES_REQUESTED** [review]
I would move this test too nova-operator/test/functional/nova_compute_ironic_controller_test.go or even make it a part of "generated configs successfully" test there

**Review by mrkisaolamb — APPROVED** [review]
lgtm! thanks

**Review by gibizer — COMMENTED** [review]
Thanks @auniyal61

**Pull request merged**